src/views/ApartmentApp/components/StudentApplication/components/ApartmentHeader/index.js

```diff
@@ -8,83 +8,47 @@ const ApartmentHeader = ({
   userProfile,
   onShowApplication,
 }) => {
-  const newApplicationContent = (
-    <React.Fragment>
-      <Grid item xs={6} sm={8}>
-        <Typography variant="body1">
-          You are not on any applications for the current semester.
-          <br />
-          Click the button to begin a new application.
-        </Typography>
-      </Grid>
-      <Grid item xs={6} sm={4}>
-        <Button
-          variant="contained"
-          onClick={onShowApplication}
-          color="primary"
-          fullWidth
-          disabled={applicationCardsOpen}
-        >
-          Create a new application
-        </Button>
-      </Grid>
-    </React.Fragment>
-  );
-
-  const editApplicationContent = (
-    <React.Fragment>
-      <Grid item xs={6} sm={8}>
-        <Typography variant="body1">
-          You are on an existing application for this semester.
-        </Typography>
-      </Grid>
-      <Grid item xs={6} sm={4}>
-        <Button
-          variant="contained"
-          onClick={onShowApplication}
-          color="primary"
-          fullWidth
-          disabled={applicationCardsOpen}
-        >
-          Edit your application
-        </Button>
-      </Grid>
-    </React.Fragment>
-  );
-
-  const viewApplicationContent = (
-    <React.Fragment>
-      <Grid item xs={6} sm={8}>
-        <Typography variant="body1">
-          You are on an existing application for this semester, but you are not assigned as the
-          editor.
-          <br />
-          Only the application editor may edit the application.
-        </Typography>
-      </Grid>
-      <Grid item xs={6} sm={4}>
-        <Button
-          variant="contained"
-          onClick={onShowApplication}
-          color="primary"
-          fullWidth
-          disabled={applicationCardsOpen}
-        >
-          View your application
-        </Button>
-      </Grid>
-    </React.Fragment>
-  );
+  const headerContent = !applicationID
+    ? {
+        primaryText: 'You are not on any applications for the current semester.',
+        secondaryText: 'Click the button to begin a new application.',
+        buttonLabel: 'Create a new application',
+      }
```

Comment:
Now that I look at it this way, `primaryText`, `secondaryText`, and `buttonLabel` seem more like props. Perhaps you could extract the content logic and make this component purely presentational? That might also make it possible to meld into the `BottomBar` more smoothly.

Reply:
On the other hand, doing so would add a lot more lines of logic in the parent component, it becomes a tradeoff between abstraction vs. modularity
In it's current concept, the StudentApplication tells the bar what the state of the application is using a few simple abstracted props, and then this component display whatever is appropriate for that state, so the component structure of the parent (StudentApplication) is more simplified and easier to understand